NagorniyAleksandr/SQLTask/src/main/java/university/dao/QueryCreatorImpl.java

```diff
@@ -0,0 +1,401 @@
+package university.dao;
+
+import org.apache.log4j.Logger;
+import university.dao.crud.CRUDQuery;
+import university.exceptions.*;
+import university.jdbc.DBConnector;
+import university.models.*;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+
+import static university.dao.converter.ToObjectConverter.*;
+
+public class QueryCreatorImpl implements QueryCreator {
+
+    private CRUDQuery crudQuery;
+    private DBConnector dbConnector;
+
+    private static final Logger log = Logger.getLogger(QueryCreatorImpl.class);
+
+    public QueryCreatorImpl(DBConnector dbConnector, CRUDQuery crudQuery) {
+        this.crudQuery = crudQuery;
+        this.dbConnector = dbConnector;
+    }
+
+    @Override
+    public List<Student> getStudentsList(int offset, int length) throws AppDBException {
+
+        try (Connection connection = dbConnector.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(
+                     "SELECT students.id, students.name, students.group_id, " +
+                             "groups.name " +
+                             "FROM students " +
+                             "LEFT JOIN groups " +
+                             "ON students.group_id = groups.id " +
+                             "LIMIT ? OFFSET ?")) {
+
+            preparedStatement.setInt(1, length);
+            preparedStatement.setInt(2, offset);
+
+            ResultSet resultSet = preparedStatement.executeQuery();
+
+            return getStudentsAsListFromResultSet(resultSet);
+        } catch (SQLException e) {
+            log.error(e.getMessage());
```

Comment:
Disagrees. I catch SQLException and wrap them in my exceptions. And I would like to see in the log 
1. Exception(SQL Exception) in the class XXX with message "YYY", 
2. Throw new Exception(My exception) from class XXX. 
Then from logs I'll be able track "from->where" my exceptions fly.